pkg/virtctl/vmexport/vmexport.go

```diff
@@ -79,55 +92,68 @@ const (
+	ErrRequiredExportType = "need to specify export kind when attempting to create a VirtualMachineExport [--pvc|--vm|--snapshot]"
+	// ErrSupportedExportType serves as error message when a supported export type is not supplied
+	ErrSupportedExportType = "unable to create Virtual Machine export for unsupported source type, please supply [--vm|--snapshot]"
 	// ErrIncompatibleExportType serves as error message when an export kind is provided with an incompatible argument
-	ErrIncompatibleExportType = "Should not specify export kind"
+	ErrIncompatibleExportType = "should not specify export kind"
+	// ErrIncompatibleExportTypeManifest serves as error message when a PVC kind is defined when getting manifest
+	ErrIncompatibleExportTypeManifest = "cannot get manifest for PVC export"
+	// ErrInvalidValue ensures that the value provided in a flag is one of the acceptable values
+	ErrInvalidValue = "%s is not a valid value, acceptable values are %s"
 
 	// progressBarCycle is a const used to store the cycle displayed in the progress bar when downloading the exported volume
 	progressBarCycle = `"[___________________]" "[==>________________]" "[====>______________]" "[======>____________]" "[========>__________]" "[==========>________]" "[============>______]" "[==============>____]" "[================>__]" "[==================>]"`
 )
 
 var (
 	// Flags
-	vm           string
-	snapshot     string
-	pvc          string
-	outputFile   string
-	insecure     bool
-	keepVme      bool
-	shouldCreate bool
-	volumeName   string
-	ttl          string
+	vm                   string
+	snapshot             string
+	pvc                  string
+	outputFile           string
+	insecure             bool
+	keepVme              bool
+	shouldCreate         bool
+	includeSecret        bool
+	exportUrl            string
+	volumeName           string
+	ttl                  string
+	manifestOutputFormat string
 )
 
 type exportFunc func(client kubecli.KubevirtClient, vmeInfo *VMExportInfo) error
 
 type HTTPClientCreator func(*http.Transport, bool) *http.Client
 
-type exportCompleteFunc func(kubecli.KubevirtClient, *VMExportInfo, time.Duration, time.Duration) error
+type exportCompleteFunc func(kubecli.KubevirtClient, *VMExportInfo, time.Duration, time.Duration, string) error
 
 // ExportProcessingComplete is used to store the function to wait for the export object to be ready.
 // Useful for unit tests.
 var ExportProcessingComplete exportCompleteFunc = waitForVirtualMachineExport
 
 type VMExportInfo struct {
-	ShouldCreate bool
-	Insecure     bool
-	KeepVme      bool
-	OutputFile   string
-	VolumeName   string
-	Namespace    string
-	Name         string
-	ExportSource k8sv1.TypedLocalObjectReference
-	TTL          metav1.Duration
+	ShouldCreate  bool
+	Insecure      bool
+	KeepVme       bool
+	IncludeSecret bool
+	OutputFile    string
+	VolumeName    string
+	Namespace     string
+	Name          string
+	OutputFormat  string
+	exportURL     string
```

Comment:
why this private unlike everything else?

Reply:
No reason, I will make it public.